### ISSUE TITLE
Solo selected columns toggle

### ISSUE THREAD

**Issue by ladlon** [turtletooth/tahoma2d#276]
I think it would be great to have the ability to quickly toggle a solo mode for display of columns, so you could quickly isolate a column so you can work on it, unobscured.

Granted, you can solo a single drawing by clicking on it in the Level strip, but sometimes you'll want a certain selection of columns visible, while others are hidden, as you work on various aspects of a drawing (where you need to see it in relation to some other elements in the scene), or are just arranging things.

Everyone will have their own reasons for using the tool... be it decluttering, toggling between two different configurations of hidden/visible elements, etc.

While a single column solo feature (which is often triggered in other programs via a modifier key + click on column eye icon) would be good, the ability to go into solo mode, and essentially arrange an alternative pattern of visible/hidden icon states, which you can then easily toggle between by clicking a solo button (as well as being able to assign a hotkey, please!) would be really great.

**Comment by manongjohn** [comment]
Right-click the Camera Stand button on the column you want to keep and select option to disable all others. Is that what you mean? I don't think its tied to any shortcuts though.

**Comment by ladlon** [comment]
Hi, manongjohn.  That brings up a menu that can be used to change the pattern of hidden/visible columns, but for one thing, it's destructive (makes changes, in some of the cases, that you cannot revert back from).

In a scenario where you have ALL columns visible, then you could get by using the 'toggle states' (hidden becomes visible, and vice versa)... where applying it a second time restores the original state... and with Show Only This, where later selecting Show All would restore the previous state....    But, if you have a random assortment of hidden/visible columns, there is no way to restore your previous pattern.

Even with that said, it still would be great to be able to go into a Solo mode, set the columns to various hidden/visible states, and be able to toggle between that and your original configuration... Kind of like a snapshot, of alternate view configurations.

Maybe one is one character... and the other is another... or certain layers that obscure the main fiew... Again, everyone will probably have a differnt use for the feature.

I imagine you could work around this by putting various things into SubLevels... but that doesn't allow for fluid, frequent alteration of those groupings... plus, it would still involve expanding/collapsing those sublevels, and then clicking on the hide/show icons for each of them... compared to a single click to toggle between your view configurations.

**Comment by manongjohn** [comment]
> but for one thing, it's destructive (makes changes, in some of the cases, that you cannot revert back from).

Eh? How so? That menu only toggles Camera visibility and whether or not to display the column in the viewer.  Images should not be modified as a result of hiding showing the column.  If it is, that is likely a bug with something.  It would help if you can show the scenario in which that happens so it can be investigated further.

If you are in timeline mode, the icon at the very top  acts as Toggle All buttons, switching the state for all levels in the timeline. So if you have some on some off, hitting the top icon will reverse their status.

All those menu options can be assigned keyboard shortcuts as well.

Btw, don't get me wrong as I can see how some of the suggestions might be helpful. I just think some of things you mention are already there. It can certainly be improved upon.

**Comment by ladlon** [comment]
Hi.  By destructive, I mean that it makes 'permanent' alterations to the status of the visible/hidden switches (not that it alters/damages anything).

Ya, you can reverse states, if you have random levels visible/hidden... but using the system I'm suggesting, you could toggle between two entirely unrelated patterns of hidden/visible.  The current system is only good for situations where everything is visible (and so you can restore it by doing a 'Show All'), or your intended configuration is the reverse of the current one... any deviations from that are not addressed with the current 'non-snapshot' system.

Even better would be if it allowed more than one snapshot, so you could have multiple show/hide configurations that you could instantly toggle between with a single click.  And, to edit any of them, you just go to a snapshot, change whatever hide/show toggles you want, and without having to click any save button, those are retained in that snapshot for when you return to it again.
